lib/chess/pieces/piece.rb

```diff
@@ -0,0 +1,116 @@
+require_relative '../chess'
+require_relative '../step'
+require 'colorize'
+require 'colorized_string'
+require 'byebug'
+
+module Chess
+  class Piece
+    attr_reader :symbol, :team, :current_step
+
+    Vector = Struct.new(:enabled, :deltas)
+
+    def initialize(position_algebraic, symbol, team, captured = false)
+      @current_step = Step.new(position_algebraic)
+      @symbol = symbol
+      @captured = captured
+      @team = team
+      @enemies_team = ([WHITE_TEAM, BLACK_TEAM] - [@team]).first
+    end
+
+    def move_to(target_position_algebraic, position_deltas_vectors, occuped_cells = nil)
+      possible_step = Step.new(target_position_algebraic)
+
+      possible_positions = possible_positions(position_deltas_vectors, occuped_cells)
+      return unless possible_positions.include?(possible_step.position.coordinates.to_a)
+
+      possible_step.add_previous_step(@current_step)
+      @current_step = possible_step
+
+      @current_step.position
+    end
+
+    def can_move_to?(target_position_algebraic, position_deltas_vectors, occuped_cells = nil)
+      possible_step = Step.new(target_position_algebraic)
+
+      possible_positions = possible_positions(position_deltas_vectors, occuped_cells)
+      possible_positions.include?(possible_step.position.coordinates.to_a)
+    end
+
+    def roll_back_step
+      @current_step = @current_step.prev_step
+      @captured = false
+
+      return unless @first_move == false
+
+      @first_move = true
+    end
+
+    def position
+      @current_step.position
+    end
+
+    def captured?
+      @captured
+    end
+
+    def to_s(piece_string)
+      return " #{piece_string} ".light_white if @team == WHITE_TEAM
+
+      " #{piece_string} ".black if @team == BLACK_TEAM
+    end
+
+    private
+
+    def change_captured_status
+      @captured = true
+    end
+
+    def possible_positions(position_deltas_vectors, occuped_cells)
```

Comment:
Since `position_deltas_vectors` is really an attribute of the classes that inherit from Piece, then you shouldn't need to pass them as arguments, just access it directly
```suggestion
    def possible_positions(occuped_cells)
```